### ISSUE TITLE
From training a model to inferring on it. 

### ISSUE THREAD

**Issue by Mlallena** [speechbrain/speechbrain#905]
Well, after the issue with the installation (now closed) I have another problem.

I am trying to create a model that distinguishes between only a few languages (instead of the 20+ languages the CommonLanguage recipe can do) and I've managed to train the model. However, when I try to use the model to check if it's working I get this error:
```
Traceback (most recent call last):
  File "pruebaIdioma.py", line 3, in <module>
    classifier = EncoderClassifier.from_hparams(source="./results/ECAPA-TDNN/2345")
  File "/home/ubuntu/gitfolder/speechbrain/speechbrain/pretrained/interfaces.py", line 248, in from_hparams
    return cls(hparams["modules"], hparams, **kwargs)
  File "/home/ubuntu/gitfolder/speechbrain/speechbrain/pretrained/interfaces.py", line 528, in __init__
    super().__init__(*args, **kwargs)
  File "/home/ubuntu/gitfolder/speechbrain/speechbrain/pretrained/interfaces.py", line 94, in __init__
    raise ValueError(f"Need modules['{mod}']")
ValueError: Need modules['mean_var_norm']
```
Checking the hyperparameters.yaml file in the folder that contains the model files, I see that it has a module called 'mean_var_norm_input', but not a 'mean_var_norm' module. The same happens with the train_ecapa_tdnn.yaml file in the hparams folder. Do I have to modify train_ecapa_tdnn.yaml so the model returns the expected 'mean_var_norm' module in the hyperparams.yaml file?

**Comment by TParcollet** [comment]
Hi! Pretrained interfaces are given to uniformize the models. For instance our users may come with 100 different language ID models using transformer, transducer, attention-based, RNN, MLP whatever. Hence, to use the interfaces, your yaml must comply to certain rules. These rules are always given at the beginning of each interface. In your case:

```python
MODULES_NEEDED = [
        "compute_features",
        "mean_var_norm",
        "embedding_model",
        "classifier",
    ]
``` 

It is important to have a separated yaml  for production. But you need all these keys in your yaml.

**Comment by Mlallena** [comment]
I think you have misunderstood my question.

The instructions that indicate how to train a new model point to using train_ecapa_tdnn.yaml. Said YAML file has the module "mean_var_norm_input", which is replicated in the YAML file produced by the training. However, when I try to use the model for the purpose of language detection, it returns the above mentioned error, which means that the YAML file the training produces is either _erroneous_ or _incomplete_.

This, as far as I can tell, indicates that the instructions are also incomplete, because they only indicate how to _train_ a model, but not how to _use_ that same model later. The only example you provide is for a model whose YAML file is modified to fit with what EncoderClassifier.from_hparams requires.

I am not asking for extra-detailed instructions - but if they indicated how to modify the results of the model training so they can be used later (something as basic as "After training, you must do {this} and {this} and {this} to use it"), it would certainly be welcome.

**Comment by TParcollet** [comment]
I am still unsure about the problem here. Could you please point me towards what you call **the instructions**? 

It is is, in fact, impossible to come with a tutorial answering the use of interfaces because they all are both model and task dependent. The most detailled instructions with have to move from training to inference is for ASR (see Step 5): https://colab.research.google.com/drive/1aFgzrUv3udM_gNJNUoLaHIm78QHtxdIz?usp=sharing

**Comment by Mlallena** [comment]
If you go here (https://github.com/speechbrain/speechbrain/tree/develop/recipes/CommonLanguage/lang_id) it indicates how to train an ECAPA-TDNN model. That part is no problem.

What would the next step be? How do I _use_ the resulting model?

**Comment by TParcollet** [comment]
Ok, this is what I explained above. Maybe the README is a bit misleading here. Look at what I sent you for ASR. The idea is just to transform your training yaml into an "inference" or "production" yaml. In this YAML you will need to comply with the MODULES_NEEDED list -> Hence have these modules in a modules variable EXACTLY with these names. If you go to HuggingFace, and to an ecapa-tdnn model, you can have a look at the hyperparams.yaml that we created so they can be used for inference. It's pretty simple to do so once you are used to it. Few variable names to change and that's it.

**Comment by Mlallena** [comment]
OK, that finally works. I've managed to both train a model and then infer on it with different audios.

My model, unfortunately, is quite prone to think that audios in Spanish are actually in Catalan (I'm only using 3 languages since it's all I need), but hopefully I'll find out how to fix that (curiously, the first model I made was more accurate, but then following epochs lost that accuracy). Perhaps with different audios I'll be able to fix it.

Thank you for your help.
